### ISSUE TITLE
Total Overhaul of shoesrb.com

### ISSUE THREAD

**Issue by JesseHerrick** [shoes/shoesrb.com#52]
I'm proposing a serious overhaul to shoesrb.com.

The current version of the website uses an old default Bootstrap layout that just looks ugly. With Shoes 4 well along its way, I think that we should start work on a new and improved http://shoesrb.com. There are a [bunch of issues](https://github.com/shoes/shoesrb.com/issues?state=open) that could be fixed with a new design.

A few things I'd like to add/improve: 
- Design Overhaul - Get rid of this version of Bootstrap. I would also be on board with getting rid of Bootstrap entirely. Just, you know, making it all :lipstick:.
- Fix the Download System - Still under discussion: #41, #34, etc.
- Organize the Code - Issues and pull requests for this site make the code harder to read and keep organized. We will need to make sure that the new site is organized.
- Documentation - The documentation for Shoes 4 will need to be a part of the new design.

I think that this would be a great addition to the impending release of Shoes 4. I would be happy to work on this extensively. Let me know what you all think.

**Comment by PragTob** [comment]
Good morning!

Thanks for your interest and suggestions :-)

Ah if only the release was impending... sure there will be an alpha release but we already got tons of rc1 issues lined up that's not even counting the ones that hopefully will be reported with the alpha :) The shoes4 docs aren't ready yet as well (planned for rc1) - we could make a new & better layout for the manual though.

I wouldn't want to feature the alpha release as the primary thing on the website. A blog post sure. A link on the download page to try new stuff out? Yeah sure as well :) But not the main thing. Dunno how the others feel.

I full heatedly agree that sorting the issues out before the new release would be great/needed! As for the design yes some sub pages need some work although for instance I really like the landing page. Things could always be approved upon though :)

 I don't think that the old bootstrap looks ugly - I like it. Plus I don't know if I have a screenshot around of what we had before :o 

I'd love to get the responsive style sheet working. Also a new version of bootstrap or something else would be cool. Personally a bit pro on bootstrap as it's the only css something framework I know and know its responsive features pretty well - open to anything though :)

cc: @wpp @wasnotrice @jasonrclark 

So thanks so much for opening this up and offering to work on it :heart: :+1:

**Comment by wpp** [comment]
@JesseHerrick thanks for bringing this up. I'm on board with the idea and had something similar brew  in my mind every time I visit one of these (I just don't have the design skills):
- http://pow.cx
- http://jekyllrb.com
- http://brew.sh

> Design Overhaul - Get rid of this version of Bootstrap. I would also be on board with getting rid of Bootstrap entirely. Just, you know, making it all :lipstick:.

Do you want to replace Bootstrap with another framework (such as foundation) or just write all the css by hand? I'm all for the :lipstick: but as far as I remember @steveklabnik is the judge of whats hot or not :smile:

> Organize the Code - Issues and pull requests for this site make the code harder to read and keep organized. We will need to make sure that the new site is organised.

Not sure what you mean by that.

> Documentation - The documentation for Shoes 4 will need to be a part of the new design.

Like @PragTob said: "The shoes4 docs aren't ready yet as well (planned for rc1)"

So in general I'm :+1: maybe you should fork the existing project and start with ripping out bootstrap and making a new design. Then make a post about your progress?

Unfortunately I don't have the time to really help out with this. But it seems like @JesseHerrick is motivated enough to pull this off.

**Comment by JesseHerrick** [comment]
@PragTob @wpp I'm a big advocate of hand coding websites rather than using a framework. I feel that it looks nicer (given that it is designed nicely) and seems less cookie-cuttery. For responsive grids, I use [gridism](https://github.com/cobyism/gridism).

I also understand that the docs and Shoes 4 are nowhere near ready, however I'd like to build a design that can accommodate them when they're ready. I understand that I am proposing a major change that not many people have time to work on, so I'll fork this project and begin work like @wpp said.

**Comment by wasnotrice** [comment]
Sounds like a fine idea to me. The current version was a big improvement over the previous version...if we can make it any better, that's great.

One weakness of the current site is that the backend is written in a not-very-accessible framework, which makes it less likely for folks to get involved. I'd vote for keeping things as simple as possible in terms of tools used.

**Comment by PragTob** [comment]
@wasnotrice backend? Afaik we don't have any backend, it's just jekyll generated static pages or am I missing something?

**Comment by steveklabnik** [comment]
I am happy to let however actually does the work do it in the way that they prefer.

**Comment by wasnotrice** [comment]
@pragtob I'm thinking of the event machine backend. Apparently I haven't contributed to the site for a while :/.

Like I was saying, the backend uses a pretty straightforward and popular framework. I like that, now that I know about it :)

**Comment by JesseHerrick** [comment]
Well, Jekyll just went 2.0.

**Comment by PragTob** [comment]
Oh sure yes it did. So an update would be nice as well I guess :-)

**Comment by PragTob** [comment]
@JesseHerrick just FYI there is gonna be a shoes4-pre1 release tomorrow-ish if nothing goes wrong, that's alpha and hardly stable, but still. No pressure just FYI :) Thanks for taking up this task, happy to see what you'll come up with. Also happy that jekyll 2.0 will support Sass.

**Comment by PragTob** [comment]
Maybe relevant/interesting: https://github.com/blog/1833-github-pages-3 (new github pages features although that'd make it harder to move away from them as pointed out on twitter) :)

**Comment by JesseHerrick** [comment]
Hey, quick question. What would be a good description for the shoes website? I'm talking like SEO type stuff.

**Comment by PragTob** [comment]
hm you mean like keywords? ruby, gui, simple, ease, fun, native applications, toolkit, gui toolkit, windows, mac, Linux, cross platform, app, beginner, packaging...

just off the top of my head

**Comment by JesseHerrick** [comment]
No, I meant like a tagline sort of thing. Like: Shoes is a tiny GUI kit that allows you to create Ruby desktop apps... (something like that)

**Comment by PragTob** [comment]
Erm. Good question!

@wasnotrice @jasonrclark @jrgifford @plexus @steveklabnik @ashbb @ccoup @KCErb @pjfitzgibbons can you think of something clever here? :)

I turned to the good old [Nobody Knows Shoes](http://cloud.github.com/downloads/shoes/shoes/nks.pdf) to see what our "spiritual father" might have to say about this.

On page 16 there is: 

> Shoes is for everyone.

Personally I'd go for something like:

> Shoes is a GUI toolkit allowing you to easily write GUI programs in Ruby

or something along those lines as you already suggested.

Here is one sentence @jasonrclark cooked up which I like (slightly modified):

> Shoes is a tiny GUI toolkit for writing fun, simple applications in Ruby

**Comment by jrgifford** [comment]
I like

> Shoes is a tiny GUI toolkit for writing fun, simple applications in Ruby

**Comment by davorb** [comment]
I prefer the first one. It's simple and straight to the point.

> Shoes is a GUI toolkit allowing you to easily write GUI programs in Ruby

**Comment by JesseHerrick** [comment]
Hey everyone! It's been a few months since this was last discussed, so I'd like to say that this has not been abandoned. I've been really busy, but I'll try to put more work into the new design.

**Comment by PragTob** [comment]
Thanks sounds cool :+1: Please don't stress out over it or anything and let us know where you need help :)

**Comment by wpp** [comment]
@JesseHerrick are you still working on this? There is some talk in other PR's about upgrading Bootstrap and FontAwesome. I have a long weekend ahead of me but don't want start anything considering you are working on a new version.

**Comment by JesseHerrick** [comment]
@wpp I saw the PR. Those changes are fine with me. I don't have too much time on my hands to work on the new site.

**Comment by nathancarnes** [comment]
@wpp @JesseHerrick my thought is to move things over to the newest version of Bootstrap to help facilitate implementing the new design. 

If someone else is already working on that upgrade, I'm happy to pitch in if I can be useful. Otherwise, let me know and I'll get going on a PR. :thumbsup:

**Comment by wpp** [comment]
@nathancarnes go for it!

**Comment by JesseHerrick** [comment]
Hey everyone! This is a **really early** view of my work on the redesign. Let me know what you think? (colors? margins? whatever)

Also, I'm not sure on the large "Shoes" header's color, so tell me if you think another color would be better.

![screen shot 2014-08-16 at 10 30 37 pm](https://cloud.githubusercontent.com/assets/3150092/3943596/fd1ef720-25b6-11e4-81c2-9a3dcc97f1b5.png)

**Comment by JesseHerrick** [comment]
And another to show responsiveness.

![screen shot 2014-08-17 at 12 19 15 am](https://cloud.githubusercontent.com/assets/3150092/3943750/c0ef0600-25c5-11e4-96bf-44b9ed48393f.png)

**Comment by jasonrclark** [comment]
That looks really sharp! I agree that the red doesn't quite seem right to me, but not positive what to use instead. The plain black in the responsive shots looks like an easy option to me, but I don't know if we're looking to give Shoes a more identifiable look with some other color scheme.

Thanks for your work on this. Super awesome! :rocket: :sparkles:

**Comment by steveklabnik** [comment]
I like it! :D

**Comment by nathancarnes** [comment]
@JesseHerrick I really like where this is headed! Nice job. 

It feels like we could do more to make the design more unique to Shoes, though. I wonder if it's worth exploring incorporating some hand-drawn elements to tie it back to the [book](https://cloud.github.com/downloads/shoes/shoes/nks.pdf). It's also possible that once there's a strong logo in place, that'll do the trick.

**Comment by JesseHerrick** [comment]
I think a logo will definitely fix it @nathancarnes. I've flipped through the book and I see what you mean, but I don't really think they would fit on the home page. Maybe in the tutorial section. We'll see.

**Comment by wpp** [comment]
Wow @JesseHerrick this looks slick. I agree that a logo and some uniqueness might be nice. Very impressive! :100:

**Comment by PragTob** [comment]
Looks good thanks so much @JesseHerrick ! :-)

I know it's WIP but I'd go with a smaller sample on the landing page (max. 20 lines) and some screenshots of the resulting app. Really like the look - keep it up :-)

Tobi

**Comment by JesseHerrick** [comment]
That makes sense @PragTob, I just typed up the stuff that was in the old page. I'll make sure to change that.

**Comment by JesseHerrick** [comment]
@PragTob, would this be too short?

``` ruby
Shoes.app(title: "Welcome to Shoes") do
  para "Hello, world!"
end
```

If so, could you suggest a short/simple code example to feature?

**Comment by KCErb** [comment]
How about something like samples 21 or 56?

**Sample 21**

``` ruby
Shoes.app width: 300, height: 100 do
  background forestgreen, curve: 30
  border gold, strokewidth: 5, curve: 30
  subtitle fg(strong('Shoes Shoes Shoes'), white), align: 'center'
end
```

![sample21](https://cloud.githubusercontent.com/assets/4992244/3957717/46627d8c-2718-11e4-86c3-5894551c0999.jpg)

**Sample 56**

``` ruby
# sample56.rb
Para = Shoes::Para
Shoes.app :width => 200, :height => 160 do
  style Para, :align => 'center', :stroke => pink, :size => 30
  stack do
    para "hello"
    para "hello", :size => 10, :stroke => red
    para "hello"
  end
end
```

![sample56](https://cloud.githubusercontent.com/assets/4992244/3957725/585232ee-2718-11e4-87b1-63daa38c694e.jpg)

Of course you may want to change them a bit (I'd change the positioning on 21 and the words on 56 for example).

**Comment by JesseHerrick** [comment]
@KCErb no on the first one. It is way too ugly to go on a nice web design. Maybe the second one.

**Comment by KCErb** [comment]
Yeah, haha :blush:. I do mean for you to clean up the prettiness a bit!

The point of my suggestion is to find a balance between something boring like
![boring](https://cloud.githubusercontent.com/assets/4992244/3957853/a704879c-2719-11e4-9e58-5d57819b2dc0.jpg)

and something that's too complicated like the current option.

I'm also in favor of something that uses more gui-ish elements like button or edit line. But again we'd want to pretty it up with some colors, centering etc.

Thanks for all your work on this @JesseHerrick!

**Comment by PragTob** [comment]
A sample that I like to use to show the simpleness of shoes is the following:

``` ruby
Shoes.app title: 'Hello Shoes' do
  background gradient limegreen..blue
  stack do
    para 'This is just a very basic app'
    button 'Click me' do alert 'Hello there!' end
    image 'http://shoesrb.com/img/shoes-icon.png'
  end
end
```

![screenshot from 2014-08-18 23 27 23](https://cloud.githubusercontent.com/assets/606517/3958283/8493388e-271e-11e4-959c-f127c98ba5e9.png)

Color scheme/could/should be updated to fit the layout.

We also could add other elements like edit lines/rectangles to show a lot of things that shoes is good for.

Thanks for your work!

Edit: notably we could make a headline that says "HELLO SHOES" with the logo beside it :)

**Comment by JesseHerrick** [comment]
Just as an FYI, here's kind of what the code example and app will look like on the site.

![screen shot 2014-08-18 at 5 28 14 pm](https://cloud.githubusercontent.com/assets/3150092/3958320/c1252816-271e-11e4-88e3-a01066a31a9d.png)

So obviously, we can't really go with the basic `para "Hello, world!"`, but I think something short and sweet that shows just what Shoes can do would be nice. I'm thinking similar to what [Ruby has](https://www.ruby-lang.org/en/) on its home page.

**Comment by eliduke** [comment]
Hey! What's the story with this issue? I'm happy to help with a redesign. Let's DO THIS!

**Comment by PragTob** [comment]
:) You'd need to get in touch with @JesseHerrick to see what he has done so far or venture on by yourself, if that's ok for Jesse.

I also played a bit more with github pages again after the jekyll 2.0 release. They now support scss/sass and other cool stuff :)

**Comment by eliduke** [comment]
Hey @JesseHerrick, what's the status of the shoesrb.com overhaul? Are you at a place where you could push something here and I could take a look at it? Let's DO THIS.

**Comment by PragTob** [comment]
Btw. @wpp  how about you?

**Comment by JesseHerrick** [comment]
@eliduke @PragTob I haven't had too much time to work on it as of late. But I have time now and will spend the rest of the day furthering the redesign.

**Comment by JesseHerrick** [comment]
Here's the latest: https://github.com/JesseHerrick/shoesrb.com

**Comment by jasonrclark** [comment]
Liking what I'm seeing so far. Thanks for the work on it @JesseHerrick!

**Comment by PragTob** [comment]
Thanks for the work @JesseHerrick - will take a look tomorrow :)

**Comment by eliduke** [comment]
Looks great, @JesseHerrick. Are there known areas that still need attention or should I just poke around and see what I can find?

**Comment by JesseHerrick** [comment]
Well @eliduke, I did a little bit of work on the contributors page, but it still needs some styles. Other than that just look around and see what you can fix. A lot of the work needed to be done is styling (which I can do if you're not up to it), but there may be some broken things also. Make sure to run the site with `bundle exec jekyll serve -w`.

I have yet to make a styleguide, but my breakpoints are done as follows:
- `morbidly-obese-bear` - for really big screens (> 1601px)
- `papa-bear` - big screens (< 1600px)
- `mama-bear` - medium screens (not used often; < 1250px)
- `goldilocks` - tablet-ish screens (< 768px)
- `baby-bear` - mobile only screens (< 650px)

You can also see a live preview at http://shoes.jesseherrick.io

**Comment by eliduke** [comment]
Sounds great. I'm happy to style that up.

**Comment by JesseHerrick** [comment]
I suppose the best way to contribute would be to fork then submit a PR to https://github.com/JesseHerrick/shoesrb.com, unless you want me to give you push rights @eliduke.

**Comment by wpp** [comment]
@PragTob how about me? I don't quite follow...

**Comment by eliduke** [comment]
@JesseHerrick I can just fork your repo and submit PRs.

**Comment by PragTob** [comment]
@wpp just wanted to ask you if you also want to do some work on a new iteration of the home page, just if you got time and motivation of course :)

Cheers and thanks everyone!!!

**Comment by eliduke** [comment]
I will be working on this for a while this morning, and I am happy to take a stab at the homepage as well.

**Comment by eliduke** [comment]
Maybe this is obvious, but I had to delete my previous fork of shoesrb.com before I was able to fork @JesseHerrick's fork. Forking a fork!

**Comment by eliduke** [comment]
@JesseHerrick I am new to gridism.css. Is there a way to set different column widths depending on screen size? I get the .half or .one-fifth syntax, but what if I want an element to be .one-fifth on a large screen but .half on mobile?

**Comment by JesseHerrick** [comment]
@eliduke I don't think you're using the grid properly. It stacks automatically on mobiles. What page are you styling?

**Comment by eliduke** [comment]
@JesseHerrick I understand that it stacks automatically, but take a look at the current contributors page:

http://shoesrb.com/contribute/

On desktop it has 6 people per row, on mobile it has 2. Is that possible with gridism? Or does everything just stack individually on mobile?

**Comment by JesseHerrick** [comment]
That can be done in CSS. If I was on a computer right now I would show you.
Take a look over at SitePoint for a grid example from a list.

**Comment by eliduke** [comment]
@JesseHerrick It's not a big deal right now. I pushed my changes so far and submitted a PR. Right now I've got 5 contributors per row and It looks great on the desktop. I think we can drop that down to 2 per row on mobile. You can show me later.

**Comment by wpp** [comment]
@PragTob Oh man I would love to. But I don't really have the time right now (around christmas is better for me). And I believe that @JesseHerrick and @eliduke are busy anyway so I don't want to zwischenfunken. Too many cooks something something :)

**Comment by JesseHerrick** [comment]
Soooooo! I got the layout nice and responsive for the contributors page, but I'm not quite sure about this effect yet. Check it out @PragTob @eliduke.

![contributors-blur](https://cloud.githubusercontent.com/assets/3150092/5325697/a2a4925c-7cc1-11e4-888b-63baa2deb85c.gif)

**Comment by PragTob** [comment]
Thanks @JesseHerrick ! You mean the blur/unblur? Seems like a playful thing to me - not must have but not bad. So I'm undecided/ok with whatever :)

(I know, not helping, sorry)

**Comment by davorb** [comment]
I think it's better without the blur.

**Comment by JesseHerrick** [comment]
Yeah @davorb, I got rid of it. I don't want users to think that they're losing their vision!
